File: examples/reticulated_python/13.py
class Fruit:
	def say(self:Fruit)->str:
		return "Fruit"
class Animal:
	def say(self:Animal)->str:
		return "animal"
class Apple(Fruit):
	def say(self:Apple)->str:
		return "apple"
class Cat(Animal):
	def say(self:Cat)->str:
		return "animal"

@fields({'f':Fruit})
class Main:
	f=Fruit()
	def main(self:Main) -> str:
		self.f = Apple()
		self.f = Cat()
		return self.f.say()

Main().main()

File: examples/strongscript/13.ts
class Fruit { say():!string {} }
class Animal { say():!string {} }
class Apple extends Fruit { say():!string { "apple" } }
class Cat extends Animal { say():!string { "cat" } }

class Main {
  main():!string {
    var f:!Fruit = new Apple()
    f = new Cat()
    f.say()
  }
}
